demo/src/app/gridster/gridster-prototype/gridster-prototype.service.ts

```diff
@@ -185,15 +191,24 @@ export class GridsterPrototypeService {
      * It checks if "element" is totally covered by "containerEl" area.
      * @param element Dragged element
      * @param containerEl Element above which "element" is dragged
+     * @param event DraggableEvent
      * @returns {boolean}
      */
-    private isInsideContainer(element, containerEl) {
-        const containerRect = containerEl.getBoundingClientRect();
-        const elRect = element.getBoundingClientRect();
-
-        return elRect.left > containerRect.left &&
-            elRect.right < containerRect.right &&
-            elRect.top > containerRect.top &&
-            elRect.bottom < containerRect.bottom;
+    private isOverGridster(item: GridsterItemPrototypeDirective, gridster: GridsterService, event): boolean {
+        const el = item.$element;
+        const elContainer = gridster.gridsterComponent.$element;
+        const tolerance = gridster.options.tolerance;
+
+        if (tolerance === 'fit') {
+            return utils.isElementFitContainer(el, elContainer);
+        }
+        if (tolerance === 'intersect') {
+            return utils.isElementIntersectContainer(el, elContainer);
+        }
+        if (tolerance === 'touch') {
+            return utils.isElementTouchContainer(el, elContainer);
+        }
+
+        return utils.isCursorAboveElement(event, elContainer);
     }
```

Comment:
```javascript
switch(tolerance) {
    case 'fit':
        return utils.isElementFitContainer(el, elContainer);
    case 'intersect':
        return utils.isElementIntersectContainer(el, elContainer);
    case 'touch':
        return utils.isElementTouchContainer(el, elContainer);
    default:
        return utils.isCursorAboveElement(event, elContainer);
}
```

Reply:
I don't like switch-case construction, but in this case it's acceptable.